youtube_dl/extractor/tver.py

```diff
@@ -0,0 +1,129 @@
+            },
+            'skip': 'Running from test_download.py doesn\'t seem to be able to handle encrypted HLS videos',
+        },
+    ]
+
+    IE_NAME = 'TVer'
+    IE_DESC = 'TVer'
+
+    _VALID_URL = r'https?://(?:www\.)?tver\.jp/(corner|episode|feature)/(?P<id>f?[0-9]+)'
+    _GEO_COUNTRIES = ['JP']  # TVer service is limited to Japan only
+
+    BRIGHTCOVE_URL_TEMPLATE = 'http://players.brightcove.net/%s/default_default/index.html?videoId=ref:%s'
+
+    def _real_extract(self, url):
+
+        # extract video id
+        video_id = self._match_id(url)
+
+        # download webpage
+        webpage = self._download_webpage(url, video_id)
+
+        # extract video information
+        video_info_csv = self._search_regex(r'addPlayer\((?P<video_info>.*?)\);', webpage, 'video information', flags=re.DOTALL).strip()
+        video_info_csv = video_info_csv.replace('\t', '').replace('\n', '').replace('\'', '')  # remove \t and \n and '
+        video_info = video_info_csv.split(',')
+
+        # extract brightcove account id
+        brightcove_account_id = video_info[3]
+
+        # extract brightcove video id
+        brightcove_video_id = video_info[4]
+
+        # brightcove url
+        brightcove_url = self.BRIGHTCOVE_URL_TEMPLATE % (brightcove_account_id, brightcove_video_id)
+
+        # debug output
+        if self._downloader.params.get('verbose', False):
+            self.to_screen('Video Information: %s' % video_info)
+            self.to_screen('Brightcove Account ID: %s' % brightcove_account_id)
+            self.to_screen('Brightcove Video ID: %s' % brightcove_video_id)
+            self.to_screen('Brightcove URL: %s' % brightcove_url)
+
+        # evacuate _VALID_URL
+        _VALID_URL = self._VALID_URL
+
+        # temporarily replace _VALID_URL
+        # prevent _VALID_URL from being the URL of Tver when executing the parent class's _real_extract () method
+        self._VALID_URL = r'https?://players\.brightcove\.net/(?P<account_id>\d+)/(?P<player_id>[^/]+)_(?P<embed>[^/]+)/index\.html\?.*(?P<content_type>video|playlist)Id=(?P<video_id>\d+|ref:[^&]+)'
+
+        # get video information
+        info_dict = super(TVerIE, self)._real_extract(brightcove_url)
+
+        # get video description
+        description = \
+            self._html_search_meta(['description', 'og:description', 'twitter:description'], webpage, 'description', default=None) or \
+            self._html_search_regex(r'<div[^>]+class="description"[^>]*>(?P<description>.*?)</div>', webpage, 'description', default=None, flags=re.DOTALL)
+
+        # undo _VALID_URL
+        self._VALID_URL = _VALID_URL
+
+        # TVer ID
+        info_dict['id'] = video_id
+        # Brightcove ID
+        info_dict['display_id'] = brightcove_video_id
+        # select large thumbnail
+        info_dict['thumbnail'] = info_dict.get('thumbnail').replace('160x90', '1920x1080')
+        # desctiption
+        info_dict['description'] = description
```

Comment:
Noway. See other extractors on how to delegate properly.

Reply:
I think I have fixed all the points you pointed out. 
Thank you for your review.